boards: shields: Use new naming scheme for shield node labels

erwango opened the pull request:
See https://github.com/zephyrproject-rtos/zephyr/issues/50040

Note: We end up with quite non nice labels like:
- `dac80508_dac80508_evm`
- `uc8176_waveshare_epaper_gdew042t2`
- `lsm303agr-magn_x_nucleo_iks01a2`

I wonder if `<shield_name>_<device>` instead of `<device>_<shield_name>` wouldn't be a better option. Toughs ?

erwango committed:
doc: porting: shields: Update instructions for nodelabel definitions

Update requirements regarding board connectors compatibility now that
dtc versions prior to 1.4.2 can't be used anymore (current required
version is 1.4.6).

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango committed:
doc: porting: shields: Provide guidelines for nodelabel

In order to avoid device definitions conflicts when compiling shields
description with boards embedding similar devices, nodelabels of devices
in shield's devicetree file should differ from the nodelabels used in
board's devicetree file.
The form to be used was discussed in #50040 and agreed to be as
<device>_<shield_name>.

Update shields documentation to make it an explicit rule.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango commented:
I've missed some nodes (ones w/o preexisting nodelabel). Kept as DNM for now.

MaureenHelm commented on doc/hardware/porting/shields.rst:
```diff
@@ -50,20 +50,9 @@ This should be done at two different level:
 
 * Devicetree: A board :ref:`devicetree <dt-guide>` file,
-  :file:`BOARD.dts` should define a node alias for each connector interface.
+  :file:`BOARD.dts` should define an alertnate nodelabel for each connector interface.
```
```suggestion
  :file:`BOARD.dts` should define an alternate nodelabel for each connector interface.
```

erwango committed:
boards: shields: fix indentation issues

Fix indentation issues in shield overlays seen when reviewing
previous commit.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango commented on doc/hardware/porting/shields.rst:
```diff
@@ -37,6 +37,18 @@ These files provides shield configuration as follows:
 
+Besides, in order to avoid name conflicts with devices that may be defined at
+board level, it is advsised, specifically for shields devicetree descriptions,
+to provide a device nodelabel is the form <device>_<address>, for instance:
```
Thanks. I actually prepared the patch sometime ago. Will fix.
I should also fix the `advsised`.

nandojve commented on boards/shields/atmel_rf2xx/atmel_rf2xx_arduino.overlay:
```diff
@@ -6,7 +6,7 @@
 / {
 	chosen {
-		zephyr,ieee802154 = &ieee802154;
+		zephyr,ieee802154_atmel_rf2xx_arduino = &ieee802154_atmel_rf2xx_arduino;
```
I didn`t understand how this will bind with samples, for instance:

https://github.com/zephyrproject-rtos/zephyr/blob/1b98db232ed32ce52d7ca9e64a7bfb44c3da0055/samples/net/wpanusb/src/wpanusb.c#L40-L41

Should samples have a overlay file?

erwango replied:
ok, my bad. `zephyr,ieee802154` should not have been changed. I'll fix it

erwango committed:
boards: shields: Update node_label according to new naming scheme

Update existing shields to conform with new node label naming scheme.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>
